server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -166,6 +173,55 @@ public Invocations getInvocations() {
       return completedFuture(builder.build());
    }
 
+   private CompletionStage<RestResponse> connectSource(RestRequest request) {
+      final NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+      builder.status(NO_CONTENT);
+
+      String cacheName = request.variables().get("cacheName");
+      ContentSource contents = request.contents();
+      byte[] config = contents.rawContent();
+
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+
+      if (cache == null) {
+         return notFoundResponseFuture();
+      }
+
+      if (config == null || config.length == 0) {
+         return badRequestResponseFuture("A remote-store config must be provided");
+      }
+
+      String storeConfig = new String(config, UTF_8);
+      Json read = Json.read(storeConfig);
+
+      if (!read.isObject() || read.at("remote-store") == null || read.asMap().size() != 1) {
+         return badRequestResponseFuture("Invalid remote-store JSON description: a single remote-store element must be provided");
+      }
+
+      ConfigurationBuilder cfgBuilder = new ConfigurationBuilder();
+      PersistenceConfigurationBuilder persistenceConfigurationBuilder = cfgBuilder.persistence();
+      RemoteStoreConfigurationBuilder storeConfigurationBuilder = persistenceConfigurationBuilder
+            .addStore(RemoteStoreConfigurationBuilder.class);
+      invocationHelper.getJsonReader().readJson(storeConfigurationBuilder, storeConfig);
+
+      Configuration configuration = cfgBuilder.build();
+      PersistenceConfiguration persistence = configuration.persistence();
+      if (persistence.stores().size() != 1) {
```

Comment:
It doesn't look like this is checked to ensure there is only a remote store? Or is that done somewhere else?

Reply:
This check is redundant, will get rid of it